File: src/integration/spring/src/jvmMain/java/community/flock/wirespec/integration/spring/java/configuration/WirespecWebMvcConfiguration.java
package community.flock.wirespec.integration.spring.java.configuration;

import community.flock.wirespec.integration.spring.java.web.WirespecMethodArgumentResolver;
import community.flock.wirespec.integration.spring.shared.WirespecJsonMapper;
import community.flock.wirespec.java.Wirespec;
import org.springframework.context.annotation.Configuration;
import org.springframework.web.method.support.HandlerMethodArgumentResolver;
import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;

import java.util.List;

@Configuration
public class WirespecWebMvcConfiguration implements WebMvcConfigurer {

    private final Wirespec.Serialization wirespecSerialization;
    private final WirespecJsonMapper wirespecJsonMapper;

    public WirespecWebMvcConfiguration(Wirespec.Serialization wirespecSerialization, WirespecJsonMapper wirespecJsonMapper) {
        this.wirespecSerialization = wirespecSerialization;
        this.wirespecJsonMapper = wirespecJsonMapper;
    }

    @Override
    public void addArgumentResolvers(List<HandlerMethodArgumentResolver> argumentResolvers) {
        argumentResolvers.add(new WirespecMethodArgumentResolver(wirespecSerialization, wirespecJsonMapper));
    }
}

File: src/integration/spring/src/jvmMain/kotlin/community/flock/wirespec/integration/spring/java/configuration/WirespecWebMvcConfiguration.kt
package community.flock.wirespec.integration.spring.java.configuration

import com.fasterxml.jackson.databind.ObjectMapper
import community.flock.wirespec.integration.spring.java.web.WirespecMethodArgumentResolver
import community.flock.wirespec.java.Wirespec
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.context.annotation.Configuration
import org.springframework.http.MediaType
import org.springframework.web.method.support.HandlerMethodArgumentResolver
import org.springframework.web.servlet.config.annotation.WebMvcConfigurer

@Configuration
open class WirespecWebMvcConfiguration : WebMvcConfigurer {

    @Autowired
    lateinit var objectMapper: ObjectMapper

    @Autowired
    lateinit var wirespecSerialization: Wirespec.Serialization

    override fun addArgumentResolvers(argumentResolvers: MutableList<HandlerMethodArgumentResolver>) {
        argumentResolvers.add(WirespecMethodArgumentResolver(objectMapper, wirespecSerialization))
    }
}
